src/TimeSpan.php

```diff
@@ -282,6 +282,29 @@ public function isPositiveOrZero(): bool
     {
         return $this->nanoseconds >= 0;
     }
+
+    public function add(self $timeSpan): self
+    {
+        return new self($this->nanoseconds + $timeSpan->nanoseconds);
```

Comment:
First add, then check that result is not float. If float, throw OutOfBoundException like in `fromX()`.